File: src/app/top-module/domain/connection-hierarchy-search-model.ts
export interface ConnectionHierarchySearchModel {
    ecckt: string;
    eccktExactMatch: boolean;
    connectionReference: string;
    connXrefExactMatch: boolean;
    locationA: string;
    locationAExactMatch: boolean;
    locationZ: string;
    locationZExactMatch: boolean;
    circuitTypes: string[];
    lookupChildren: boolean;
}

File: src/main/java/com/verizon/dashvue/top/domain/ConnectionHierarchySearchModel.java
package com.verizon.dashvue.top.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;


@Data
@NoArgsConstructor
@AllArgsConstructor
public class ConnectionHierarchySearchModel {

    private String ecckt;
    private Boolean eccktExactMatch;
    private String connectionReference;
    private Boolean connXrefExactMatch;
    private String locationA;
    private Boolean locationAExactMatch;
    private String locationZ;
    private Boolean locationZExactMatch;
    private List<String> circuitTypes;
    private Boolean lookupChildren;
}
